diesel/src/mysql/connection/raw.rs

```diff
@@ -179,6 +188,20 @@ impl RawConnection {
         unsafe { ffi::mysql_next_result(self.0.as_ptr()) };
         self.did_an_error_occur()
     }
+
+    #[cfg(not(windows))]
```

Comment:
Is this a fundamental restriction of `libmysqlclient` that ssl is not supported on windows or is this a restriction of the `libmysqlclient-sys` crate?

Reply:
It is a restriction of `libmysqlclient-sys`. There is a PR https://github.com/sgrif/mysqlclient-sys/pull/28
which would make all `not(windows)` in this once checks obsolete.

Reply:
I would rather not like to introduce such special case handling then. The current situation with mysqlclient seems to be quite unfortunate as I cannot merge PR's there. As we've hit that problem already more than once we should definitively seek a solution for this. Maybe @sgrif can grant me the corresponding rights or transfer the repositories to the diesel organization?

Beside of that as a long term goal it would probably be great to replace libmysqlclient with a pure rust implementation.

Reply:
> Beside of that as a long term goal it would probably be great to replace libmysqlclient with a pure rust implementation.

You've mentioned [mysql](https://docs.rs/mysql/19.0.1/mysql/) crate in https://github.com/sgrif/mysqlclient-sys/issues/25#issuecomment-623016021 in regard to this matter. But is performance of the crate comparable to `libmysqlclient-sys`?

Reply:
I'm not sure if the performance is comparable. That's one of the many open questions that need to be answered before such an implementation can be merged. As starting point have a look at the benchmarks I've pushed [here](https://github.com/weiznich/diesel/tree/more/benches/diesel_bench) (Improvements are welcome)

Reply:
@thomasmost That's the most important open issue. Someone with access to a windows development environment needs to update mysqlclient-sys.